src/vm/comutilnative.cpp

```diff
@@ -2624,14 +2626,107 @@ FCIMPL6(INT32, ManagedLoggingHelper::GetRegistryLoggingValues, CLR_BOOL* bLoggin
 }
 FCIMPLEND
 
+static BOOL HasOverriddenEqualsWorker(MethodTable* mt)
+{
+    MethodTable::IntroducedMethodIterator it(mt);
+    for (; it.IsValid(); it.Next())
+    {
+        if (strcmp(it.GetMethodDesc()->GetName(), "Equals") == 0)
```

Comment:
> code sounds like not comparing the signature of the method

It is checking whether the virtual slot has been overridden. It is the only 100% accurate way to check whether the method has been overridden.